addons/base_hash/models/secure_sequence_mixin.py

```diff
@@ -0,0 +1,36 @@
+from odoo import fields, models, _
+
+
+class SecureSequenceMixin(models.AbstractModel):
+    _name = 'secure.sequence.mixin'
+    _description = "Secure Sequence Mixin"
+
+    secure_sequence_id = fields.Many2one('ir.sequence',
+                                         help='Sequence to use to ensure the securisation of data',
+                                         check_company=True,
+                                         readonly=True,
+                                         copy=False)
+
+    def _get_secure_sequence(self):
+        """This method can ve overriden if the name of the secure sequence
+        does not correspond to the default one 'secure_sequence_id' as per
+        _create_secure_sequence's default parameter"""
+        return self.secure_sequence_id
+
+    def _create_secure_sequence(self, company_id, name='secure_sequence_id'):
```

Comment:
> Just always force it to be `secure_sequence_id` (you are creating the field in here anyways, so if it is not the same name, you will have a duplicate)

It cannot be the case because for instance in `l10n_fr{,pos_cert}`, we have two secure sequence in company: `l10n_fr_closing_sequence_id` and `l10n_fr_pos_cert_sequence_id`. If we always use `secure_sequence_id`, then they would collide

Reply:
> `company_id` is probably not required either Here is home the generic method `_check_company` does
> 
> https://github.com/odoo/odoo/blob/f5ff26f42e63c41143444a6b1122d71bc2c1071d/odoo/models.py#L3527

I'm not sure since I get this error (after installing `l10n_fr_pos_cert` when another loca is already installed) after removing `'company_id': company_id`
![2022-04-25_12-11](https://user-images.githubusercontent.com/91618364/165069797-12bc3f8e-195c-4409-96fa-bc803884b720.png)

Reply:
> It cannot be the case because for instance in `l10n_fr{,pos_cert}`, we have two secure sequences in company: `l10n_fr_closing_sequence_id` and `l10n_fr_pos_cert_sequence_id`. If we always use `secure_sequence_id`, then they would collide

Then you have a useless field because of the one declared in your mixin.

> I'm not sure since I get this error (after installing `l10n_fr_pos_cert` when another loca is already installed) after removing `'company_id': company_id`

What I meant is that you can get the company_id in a generic manner without providing it in the arguments of the function. Odoo is object oriented, trying to mix functional things just makes it less consistent and  a pain in the a** when we need to bugfix without changing the API.

Reply:
> Then you have a useless field because of the one declared in your mixin.

👍

> What I meant is that you can get the company_id in a generic manner without providing it in the arguments of the function. Odoo is object oriented, trying to mix functional things just makes it less consistent and a pain in the a** when we need to bugfix without changing the API.

The thing is that `self` might be an `account_journal`, a `picking_type` or a `company` (and potentially other things in the future) so `self.company_id.id` will not always work. I'm not sure what would be better between something like this (additional param), and a try/except or whatever mechanism to check whether we should access `self.company_id.id` (assuming our model always has a `company_id`) or `self.id`.
And `self.env.company.id` isn't always the same as the provided `company_id` as argument so it cannot be used here (for instance when we install `l10n_fr_pos_cert`)